### ISSUE TITLE
How to interpret the validation loss?

### ISSUE THREAD

**Issue by GoncaloCdM** [ultralytics/yolov5#1867]
## ❔Question
Despite the good results for both the best mAP epoch and the last epoch (1300th step):
![](https://i.imgur.com/WAOnlBH.png)

This is what the validation loss looks in contrast with the train loss:
![](https://i.imgur.com/G25Jxcr.png)

Any thoughts on why this is happening?

**Comment by glenn-jocher** [comment]
@GoncaloCdM overfitting. You might try increasing regularization or reducing loss component gains for overfitting loss components in your hyperparameters. Or you could evolve your hyps if you have time and resources.

**Comment by JoJoliking** [comment]
@glenn-jocher 
Dear Author:
The network structure of V4 version mainly uses C3 model and SiLU activation function. I would like to ask the author a question. Compared with the original CSPBottleNet, why do these two components bring benefits? Especially in C3 module, only the position of Conv function is changed.
Thank you.

**Comment by glenn-jocher** [comment]
@JoJoliking why do the changes bring benefits? I think in ML in general the empirical results are always more or less their own answer. Theory only goes so far in this field.

In general though eliminating convolutions is not an accuracy improver, it's a speed improver.

**Comment by JoJoliking** [comment]
@glenn-jocher 
Thank you. So I will also do some experiment to verify it.
Would you like to share your experience in parameter adjustment or experiment ？. Or will this part be reflected in the paper which you are going to publish (like yolo4).
I think this will be benefit me a lot !

**Comment by glenn-jocher** [comment]
@JoJoliking yes we will add these details to the paper. For hyperparameter adjustments for custom models see https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution

**Comment by GoncaloCdM** [comment]
> @GoncaloCdM overfitting. You might try increasing regularization or reducing loss component gains for overfitting loss components in your hyperparameters. Or you could evolve your hyps if you have time and resources.

@glenn-jocher 
Thank you. I will try that right away.
But could you please elaborate on this? YOLOv5 seems to choose the best weights based on the highest mAP. My Train/Test/Validation sets do not have any sample in common. How would you justify such good results for my Test set using the best weight (mAP) considering that the loss is increasing?

Also, would you adjust the number of steps and decrease it? Thanks so much!

**Comment by glenn-jocher** [comment]
@GoncaloCdM I can't provide any help. If you have hardware and time you can evolve hyperparameters. If you have expertise you can adjust them manually towards your desired endpoint. Good luck.

**Comment by GoncaloCdM** [comment]
@glenn-jocher OK! Will do. Thanks anyway!

**Comment by GoncaloCdM** [comment]
@glenn-jocher After some minor adjustments, it was possible to achieve better results. Thanks for your tips. I do have some concerns with `obj_loss`, however. I'm not sure I understand the behavior of the validation loss for `obj_loss`. IIRC, `box_loss` measures how accurate are the predicted BBs around the true object, `cls_loss` measures the correctness of the classification of each predicted BB, but I don't see how `obj_loss` correlates to any of those, given these results:
<img width="923" alt="Screen Shot 2021-03-01 at 5 55 23 PM" src="https://user-images.githubusercontent.com/4027419/109538356-e1c01d80-7ab7-11eb-9fe4-a765d0d56608.png">

**Comment by glenn-jocher** [comment]
@GoncaloCdM you can learn about objectness from the original YOLO papers here: https://arxiv.org/abs/1804.02767

**Comment by epratheeban** [comment]
@GoncaloCdM If your validation objective loss is increasing, then there may be two possible causes.
1. Validation set might have missed annotation 
2. Validation set might have wrong annotation
It means the objectiveness score penalizing the model for detecting wrong bounding box.
